### PR TITLE
gtk3: 3.24.18 → 3.24.20

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
Fixes: https://github.com/NixOS/nixpkgs/pull/86287

Changes: https://gitlab.gnome.org/GNOME/gtk/blob/db39ce2b1699acaf5d0ba160cf7cc68ccf6e16d6/NEWS#L1-38

Also switch to a merged pkg-config patch and add a patch comment.

cc @hedning 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/gtk/3.x.nix [hunk 1]
```diff
@@ -71,12 +71,17 @@ stdenv.mkDerivation rec {
       sha256 = "0g6fhqcv8spfy3mfmxpyji93k8d4p4q4fz1v9a1c1cgcwkz41d7p";
     })
 
-    # https://gitlab.gnome.org/GNOME/gtk/merge_requests/1002
-    ./patches/01-build-Fix-path-handling-in-pkgconfig.patch
+    # Fix path handling in pkg-config
+    # https://gitlab.gnome.org/GNOME/gtk/merge_requests/1793
```
Comment (worldofpeace) [inline pkgs/development/libraries/gtk/3.x.nix:75]: it's soo confusing that Matthias would reject it and then accept this patch later on :grin:

Comment (jtojnar) [reply pkgs/development/libraries/gtk/3.x.nix:75]: You have to present these things as a clean-up, not a compat patch. Also this one is a no-op for normal Meson, your patch made the paths always absolute.

Comment (worldofpeace) [reply pkgs/development/libraries/gtk/3.x.nix:75]: :rofl: Sneaky sneaky, make it so they don't have to think about it.

**Comment by cole-h** [comment]
@ofborg eval

"Once more unto the breach!" (Might take a while to actually start since I've done this for a few other PRs in the past few minutes.)

**Pull request merged**